File: spec/lua/test_debug_switch_user.lua
local luaunit = require("luaunit")

require("debug_switch_user")

TestDebugSwitchUser = {}

function TestDebugSwitchUser:test_debug_switch_user()
    local r = DebugSwitchUser:from_file("src/nav_parent_switch.bin")

    -- --debug implies --no-auto-read
    r:_read()

    luaunit.assertEquals(r.code, 1)
    luaunit.assertEquals(r.data.val, -190)
end

File: spec/javascript/test_debug_switch_user.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('DebugSwitchUser', 'src/nav_parent_switch.bin', function(r, DebugSwitchUser_) {
  // --debug implies --no-auto-read
  r._read();

  assert.strictEqual(r.code, 1);
  assert.strictEqual(r.data.val, -190);
});
